### PR TITLE
[Merged by Bors] - chore: renaming all `Column` to `Col`

### PR DESCRIPTION
**Pull request by madvorak**
* `Matrix.fromColumns` becomes `Matrix.fromCols`
* `Matrix.toColumns` becomes `Matrix.toCols`
* `Matrix.updateColumn` becomes `Matrix.updateCol`

Moves:
- `*fromColumns*` -> `*fromCols*`
- `*toColumns*` -> `*toCols*`
- `*updateColumn*` -> `*updateCol*`

Discussion:
https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Matrix.2EfromRows.20and.20Matrix.2EfromColumns

### CODE REVIEW

File: Mathlib/Data/Matrix/ColumnRowPartitioned.lean [hunk 1]
```diff
@@ -148,22 +148,22 @@ lemma fromRows_neg (A₁ : Matrix m₁ n R) (A₂ : Matrix m₂ n R) :
...
 @[simp]
-lemma fromColumns_fromRows_eq_fromBlocks (B₁₁ : Matrix m₁ n₁ R) (B₁₂ : Matrix m₁ n₂ R)
+lemma fromCols_fromRows_eq_fromBlocks (B₁₁ : Matrix m₁ n₁ R) (B₁₂ : Matrix m₁ n₂ R)
     (B₂₁ : Matrix m₂ n₁ R) (B₂₂ : Matrix m₂ n₂ R) :
-    fromColumns (fromRows B₁₁ B₂₁) (fromRows B₁₂ B₂₂) = fromBlocks B₁₁ B₁₂ B₂₁ B₂₂ := by
+    fromCols (fromRows B₁₁ B₂₁) (fromRows B₁₂ B₂₂) = fromBlocks B₁₁ B₁₂ B₂₁ B₂₂ := by
   ext (_ | _) (_ | _) <;> simp
 
 @[simp]
 lemma fromRows_fromColumn_eq_fromBlocks (B₁₁ : Matrix m₁ n₁ R) (B₁₂ : Matrix m₁ n₂ R)
```
Comment (eric-wieser) [inline Mathlib/Data/Matrix/ColumnRowPartitioned.lean:164]:
```suggestion
lemma fromRows_fromCols_eq_fromBlocks (B₁₁ : Matrix m₁ n₁ R) (B₁₂ : Matrix m₁ n₂ R)
```

File: Mathlib/Data/Matrix/ColumnRowPartitioned.lean [hunk 2]
```diff
@@ -34,14 +34,14 @@ def fromRows (A₁ : Matrix m₁ n R) (A₂ : Matrix m₂ n R) : Matrix (m₁ 
 
 /-- Concatenate together two matrices B₁[m × n₁] and B₂[m × n₂] with the same rows (M) to get a
 bigger matrix indexed by [m × (n₁ ⊕ n₂)] -/
-def fromColumns (B₁ : Matrix m n₁ R) (B₂ : Matrix m n₂ R) : Matrix m (n₁ ⊕ n₂) R :=
+def fromCols (B₁ : Matrix m n₁ R) (B₂ : Matrix m n₂ R) : Matrix m (n₁ ⊕ n₂) R :=
```
Comment (eric-wieser) [inline Mathlib/Data/Matrix/ColumnRowPartitioned.lean:37]: These definitions should all come with `deprecated` `alias`es that preserve the old names.

I think it's probably ok not to deprecate the lemmas, but you should list all the renames in the "Moves" section of the PR.

Comment (kim-em) [reply Mathlib/Data/Matrix/ColumnRowPartitioned.lean:37]: I have a slight preference for deprecating all the lemmas, too, but won't insist.

Comment (madvorak) [reply Mathlib/Data/Matrix/ColumnRowPartitioned.lean:37]: All right, I'll add `deprecated` everywhere.

File: Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean [hunk 3]
```diff
@@ -342,41 +342,48 @@ theorem det_updateRow_eq_zero (h : i ≠ j) :
...
-theorem det_updateColumn_smul_left (M : Matrix n n R) (j : n) (s : R) (u : n → R) :
-    det (updateColumn (s • M) j u) = s ^ (Fintype.card n - 1) * det (updateColumn M j u) := by
+theorem det_updateCol_smul_left (M : Matrix n n R) (j : n) (s : R) (u : n → R) :
+    det (updateCol (s • M) j u) = s ^ (Fintype.card n - 1) * det (updateCol M j u) := by
   rw [← det_transpose, ← updateRow_transpose, transpose_smul, det_updateRow_smul_left]
   simp [updateRow_transpose, det_transpose]
 
-@[deprecated (since := "2024-11-03")] alias det_updateColumn_smul' := det_updateColumn_smul_left
+@[deprecated (since := "2024-11-03")] alias det_updateColumn_smul' := det_updateCol_smul_left
+@[deprecated (since := "2024-11-03")] alias det_updateColumn_smul_left := det_updateCol_smul_left
```
Comment (kim-em) [inline Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean:386]: The dates in this file are incorrect.

**Comment by kim-em** [comment]
Please fix the dates, then merge.

bors d+

File: Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean (hunk 3 above)
Comment (madvorak) [reply Mathlib/LinearAlgebra/Matrix/Determinant/Basic.lean:386]: Thanks for noticing!

**Pull request closed without merging**